core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -151,6 +155,7 @@ public FuzzyQueryBuilder(String fieldName, boolean value) {
      * @param value The value of the term
      */
     public FuzzyQueryBuilder(String fieldName, Object value) {
+        DEPRECATION_LOGGER.deprecated("{} query is deprecated. Instead use the [match] query with fuzziness parameter", NAME);
```

Comment:
add a test for this? If I remember correctly the deprecation logger is hooked into the response header warnings, hence testable from the outside without looking at logs.